janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -1054,8 +1054,9 @@ public boolean supports(KeyInformation information) {
 
     @Override
     public String mapKey2Field(String key, KeyInformation information) {
-        Preconditions.checkArgument(!StringUtils.containsAny(key,new char[]{' '}),"Invalid key name provided: %s",key);
-        return key;
+        Preconditions.checkArgument(!StringUtils.containsAny(key, new char[]{ IndexProvider.REPLACEMENT_CHAR }),
```

Comment:
Can you pull this check into a method and reuse it in all indexing backends?

Reply:
Done. Though while doing this I realized that the check may not actually be necessary; I think we only need a one-way mapping from key to external indexing backend field, and having something that maps back the other way is not actually needed - in other words, we replace spaces with a special char, but we never have to replace the special char with spaces.

So I think these checks may be an abundance of caution, but it might be nice future-proofing. I'm slightly inclined to leave them as I doubt they'll inconvenience anyone, but I could go either way.